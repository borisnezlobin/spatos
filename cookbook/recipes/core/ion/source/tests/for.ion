for i in 1..6
    echo $i
end

for i in 1...5
    echo $i
end

for i in 1 2 3 4 5
    echo $i
end

for i in @(echo 1 2 3 4 5)
    echo $i
end

for i in [1 2 3 4 5]
    echo $i
end

for i in 1{A{1,2},B{1,2}}
    echo $i
end

for x y z in 1..=10
    echo $x, $y, $z
end

let array = [ 1 2 3 4 5 ]
for i in @array
    echo $i
end

let multi_line_input = "one
two
three"

for word in $multi_line_input
    echo "; $word"
end

mkdir for_test
cd for_test
touch "a b c"
touch "d e f"
touch "g"
touch "h i"
for file in *
    echo $file
end
cd ..
rm for_test -R
