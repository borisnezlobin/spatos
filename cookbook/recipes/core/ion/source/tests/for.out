1
2
3
4
5
1
2
3
4
5
1
2
3
4
5
1
2
3
4
5
1
2
3
4
5
1A1
1A2
1B1
1B2
1, 2, 3
4, 5, 6
7, 8, 9
10, , 
1
2
3
4
5
; one
two
three
a b c
d e f
g
h i
